src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,156 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(tail, value, null);
+        if (isEmpty()) {
+            head = node;
+        } else {
+            tail.next = node;
+        }
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Invalid Index");
+        }
+        if (index == 0) {
+            if (size == 0) {
+                add(value);
+                return;
+            }
+            Node<T> rightNeighbor = head;
+            Node<T> node = new Node<>(null, value, rightNeighbor);
+            head = node;
+            rightNeighbor.prev = node;
+            size++;
+        } else {
+            Node<T> tempNode = head;
+            for (int i = 0; i < index - 1; i++) {
+                tempNode = tempNode.next;
+            }
+            Node<T> leftNeighbor = tempNode;
+            Node<T> rightNeighbor = tempNode.next;
+            Node<T> node = new Node<>(leftNeighbor, value, rightNeighbor);
+            leftNeighbor.next = node;
+            if (rightNeighbor != null) {
+                rightNeighbor.prev = node;
+            } else {
+                tail = node;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (checkIndex(index) == false) {
+            throw new IndexOutOfBoundsException("Invalid Index");
+        }
+        Node<T> tempNode = head;
+        for (int i = 0; i < index; i++) {
+            tempNode = tempNode.next;
+        }
+        return tempNode.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (checkIndex(index) == false) {
+            throw new IndexOutOfBoundsException("Invalid Index");
```

Comment:
you should throw an exception in checkIndex method too, here in this method you just call checkIndex method, it should be void not boolean